node/src/components/consensus/protocols/highway.rs

```diff
@@ -415,6 +423,26 @@ impl<I: NodeIdT, C: Context + 'static> HighwayProtocol<I, C> {
             .last_finalized()
             .map_or(false, is_switch)
     }
+
+    /// Prevalidates the vertex but checks the cache for previously validated vertices.
+    /// Avoids multiplate validation of the same vertex.
```

Comment:
```suggestion
    /// Avoids multiple validations of the same vertex.
```
(Or "repeated validation", or "redundant validation".)

Reply:
Reverted in [`81c8391` (#1175)](https://github.com/CasperLabs/casper-node/pull/1175/commits/81c8391fc53aaf2512a9398393fefaba1277d608).

Reply:
No, this one I had overlooked in my first review.

Reply:
Fixed [`c23aa66` (#1175)](https://github.com/CasperLabs/casper-node/pull/1175/commits/c23aa668be091222c846512f13de3ec2db2ddcf2)